dali/operators/numba_function/numba_func.cc

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include "dali/operators/numba_function/numba_func.h"
+
+namespace dali {
+
+DALI_SCHEMA(experimental__NumbaFunc)
+  .DocStr(R"code(Invokes a compiled Numba function passed as a pointer.
+
+.. note::
+    This operator is experimental and its API might change without notice.)code")
+  .NumInput(1)
+  .NumOutput(1)
+  .AddArg("fn_ptr", R"code(Numba function pointer.
+  
+The function should be a Numba C callback function (annotated with cfunc) with the following function signature::
+
+  types.void(
+    types.CPointer(OUTPUT_DTYPE), # Pointer to output sample
+    types.CPointer(types.int64),  # Pointer to output sample shape
+    types.CPointer(INPUT_DTYPE),  # Pointer to input sample
+    types.CPointer(types.int64),  # Pointer to input sample shape
+    types.int32                   # Number of sample dimensions
+  )
+
+)code", DALI_INT64)
+  .AddOptionalArg<int>("setup_fn", R"code(Pointer to function which should return output shape.
+  
+The function should be a Numba C callback function (annotated with cfunc) with the following function signature::
```

Comment:
I would add a note, if function is not provided it is assumed that output shape and type is the same as input.